### PR TITLE
Change hash generation logic

### PR DESCRIPTION
**Pull request by lalagvana**
UserPassword lies in Models.Dto layer in order to avoid circular dependencies.

### CODE REVIEW

File: test/UserService.Mappers.UnitTests/UserCredentialsMapperTests.cs [hunk 1]
```diff
@@ -63,35 +64,12 @@ public void ShouldReturnNewDbUserCredentialsWhenDataCorrectForGetUser()
...
-                Status = "Example",
-                Password = password,
-                IsAdmin = false,
-                IsActive = true,
-                AvatarFileId = Guid.NewGuid()
-            };
+            var result = mapper.Map(userRequest);
+            result.Salt = salt;
+            result.PasswordHash = Encoding.UTF8.GetString(new SHA512Managed().ComputeHash(
+                    Encoding.UTF8.GetBytes(result.Salt + userRequest.Login + userRequest.Password + SALT3)));
```
Comment (djino97) [inline test/UserService.Mappers.UnitTests/UserCredentialsMapperTests.cs:70]: use interpolation, uses slightly less memory

File: src/UserService.Mappers/UserCredentialsMapper.cs [hunk 2]
```diff
@@ -13,18 +13,23 @@ namespace LT.DigitalOffice.UserService.Mappers
     /// </summary>
     public class UserCredentialsMapper : IMapper<UserRequest, DbUserCredentials>
     {
+        internal static string SALT3 = "LT.DigitalOffice.SALT3";
```
[the remark this answers is not in the document]
Comment (lalagvana) [reply src/UserService.Mappers/UserCredentialsMapper.cs:16]: image

File: src/UserService/UserService.csproj [hunk 3]
```diff
@@ -6,14 +6,14 @@
...
+    <PackageReference Include="FluentValidation" Version="9.2.2" />
     <PackageReference Include="GreenPipes" Version="4.0.0" />
-    <PackageReference Include="MassTransit" Version="7.0.3" />
-    <PackageReference Include="MassTransit.AspNetCore" Version="7.0.3" />
-    <PackageReference Include="MassTransit.Extensions.DependencyInjection" Version="7.0.3" />
-    <PackageReference Include="MassTransit.RabbitMQ" Version="7.0.3" />
+    <PackageReference Include="MassTransit" Version="7.0.4" />
+    <PackageReference Include="MassTransit.AspNetCore" Version="7.0.4" />
+    <PackageReference Include="MassTransit.Extensions.DependencyInjection" Version="7.0.4" />
+    <PackageReference Include="MassTransit.RabbitMQ" Version="7.0.4" />
```
Comment (djino97) [inline src/UserService/UserService.csproj:14]: remove

File: src/UserService/UserService.csproj [hunk 4]
```diff
@@ -6,14 +6,14 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="FluentValidation" Version="9.0.1" />
+    <PackageReference Include="FluentValidation" Version="9.2.2" />
```
Comment (djino97) [inline src/UserService/UserService.csproj:9]: remove

File: test/UserService.Business.UnitTests/UserService.Business.UnitTests.csproj [hunk 5]
```diff
@@ -11,12 +11,12 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="FluentValidation" Version="9.0.1" />
-    <PackageReference Include="LT.DigitalOffice.Kernel" Version="1.0.6" />
-    <PackageReference Include="Moq" Version="4.14.5" />
+    <PackageReference Include="FluentValidation" Version="9.2.2" />
```
Comment (djino97) [inline test/UserService.Business.UnitTests/UserService.Business.UnitTests.csproj:14]: remove

File: src/UserService.Mappers/UserCredentialsMapper.cs (hunk 2 above)
Comment (djino97) [inline src/UserService.Mappers/UserCredentialsMapper.cs:16]: Maybe move it to the static class and with static method generation password hash?

File: src/UserService.Business/DisableUserByIdCommand.cs [hunk 6]
```diff
@@ -36,7 +36,7 @@ public void Execute(Guid userId, Guid requestingUserId)
             repository.EditUser(editedUser);
         }
 
-        private bool GetResultCheckingUserRights(Guid userId)
+        private bool GetResultCheckingUserRights()
```
Comment (DarkFelas) [inline src/UserService.Business/DisableUserByIdCommand.cs:39]: const int rightId = 1;

return accessValidator.IsAdmin() || accessValidator.HasRights(rightId);

File: src/UserService.Business/DisableUserByIdCommand.cs [hunk 7]
```diff
@@ -22,9 +22,9 @@ public class DisableUserByIdCommand : IDisableUserByIdCommand
 
         public void Execute(Guid userId, Guid requestingUserId)
         {
-            var isAcces = GetResultCheckingUserRights(requestingUserId);
+            var isAccess = GetResultCheckingUserRights();
```
Comment (DarkFelas) [inline src/UserService.Business/DisableUserByIdCommand.cs:25]: remove isAccess variable, move GetResultCheckingUserRights() inside IF

File: src/UserService.Business/DisableUserByIdCommand.cs [hunk 8]
```diff
@@ -22,9 +22,9 @@ public class DisableUserByIdCommand : IDisableUserByIdCommand
...
 
-            if (!isAcces)
+            if (!isAccess)
             {
                 throw new Exception("Not enough rights.");
```
Comment (DarkFelas) [inline src/UserService.Business/DisableUserByIdCommand.cs:29]: Kernel ForbiddenException

File: src/UserService.Mappers/UserCredentialsMapper.cs [hunk 9]
```diff
@@ -13,18 +13,23 @@ namespace LT.DigitalOffice.UserService.Mappers
...
                 throw new ArgumentNullException(nameof(request));
             }
 
+            var salt = Guid.NewGuid().ToString() + Guid.NewGuid().ToString();
+
             return new DbUserCredentials
             {
                 Login = request.Login,
+                Salt = salt,
                 PasswordHash = Encoding.UTF8.GetString(new SHA512Managed().ComputeHash(
```
Comment (DarkFelas) [inline src/UserService.Mappers/UserCredentialsMapper.cs:31]: Create some helper for this method

File: src/UserService.Mappers/UserCredentialsMapper.cs [hunk 10]
```diff
@@ -13,18 +13,23 @@ namespace LT.DigitalOffice.UserService.Mappers
     /// </summary>
     public class UserCredentialsMapper : IMapper<UserRequest, DbUserCredentials>
     {
+        internal static string SALT3 = "LT.DigitalOffice.SALT3";
+
         public DbUserCredentials Map(UserRequest request)
         {
             if (request == null)
             {
                 throw new ArgumentNullException(nameof(request));
```
Comment (DarkFelas) [inline src/UserService.Mappers/UserCredentialsMapper.cs:22]: BadRequestException

File: test/UserService.Data.UnitTests/UserCredentialsRepositoryTests.cs [hunk 11]
```diff
@@ -17,10 +17,10 @@ class UserCredentialsRepositoryTests
...
         private IUserCredentialsRepository repository;
 
-        private DbUserCredentials firstUserCredentials;
-        private DbUserCredentials secondUserCredentials;
-        private Guid firstUserId;
-        private Guid secondUserId;
+        private Guid userId;
+        private string salt;
+        private DbUserCredentials userCredentials;
+        internal const string SALT3 = "LT.DigitalOffice.SALT3";
```
Comment (DarkFelas) [inline test/UserService.Data.UnitTests/UserCredentialsRepositoryTests.cs:23]: Remove this one from here, use 
https://docs.microsoft.com/ru-ru/dotnet/api/system.runtime.compilerservices.internalsvisibletoattribute?view=netcore-3.1

**Review by DarkFelas — CHANGES_REQUESTED** [review]

File: src/UserService.Mappers/UserCredentialsMapper.cs (hunk 2 above)
Comment (djino97) [reply src/UserService.Mappers/UserCredentialsMapper.cs:16]: And this field should be const string

**Review by DarkFelas — APPROVED** [review]

File: test/UserService.Business.UnitTests/DisablingUserByIdCommandTests.cs [hunk 12]
```diff
@@ -113,10 +110,7 @@ public void ShouldUserDisabledSuccess()
                 .Setup(x => x.EditUser(It.IsAny<DbUser>()))
                 .Returns(true);
 
-            command.Execute(userId, requestingUserId);
-
-            repositoryMock.Verify(repository => repository.GetUserInfoById(userId), Times.Once);
-            repositoryMock.Verify(repository => repository.EditUser(It.IsAny<DbUser>()), Times.Once);
```
Comment (djino97) [inline test/UserService.Business.UnitTests/DisablingUserByIdCommandTests.cs:119]: Why removed?

File: src/UserService.Models.Dto/UserPassword.cs [hunk 13]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+
+[assembly: InternalsVisibleTo("LT.DigitalOffice.UserService.Data.UnitTests")]
+[assembly: InternalsVisibleTo("LT.DigitalOffice.UserService.Mappers.UnitTests")]
+[assembly: InternalsVisibleTo("LT.DigitalOffice.UserService.Mappers")]
```
Comment (djino97) [inline src/UserService.Models.Dto/UserPassword.cs:7]: Why separated? Maybe made collapse 
image

File: src/UserService.Business/DisableUserByIdCommand.cs [hunk 14]
```diff
@@ -22,11 +23,11 @@ public class DisableUserByIdCommand : IDisableUserByIdCommand
 
         public void Execute(Guid userId, Guid requestingUserId)
         {
-            var isAccess = GetResultCheckingUserRights(requestingUserId);
+            const int rightId = 1;
```
Comment (djino97) [inline src/UserService.Business/DisableUserByIdCommand.cs:26]: Maybe move this in field of class?

Comment (lalagvana) [reply src/UserService.Business/DisableUserByIdCommand.cs:26]: I'm not sure it's really needed

File: src/UserService.Business/DisableUserByIdCommand.cs [hunk 15]
```diff
@@ -22,11 +23,11 @@ public class DisableUserByIdCommand : IDisableUserByIdCommand
 
         public void Execute(Guid userId, Guid requestingUserId)
```
Comment (djino97) [inline src/UserService.Business/DisableUserByIdCommand.cs:24]: And requestingUserId is not needed here

**Review by djino97 — APPROVED** [review]

**Pull request merged**